src/NuGet.Core/NuGet.Commands/SourcesCommands/SourceRunners.cs

```diff
@@ -336,6 +336,27 @@ public static ISettings GetSettings(string configfile)
             }
         }
 
+        public static ISettings GetSettingsFromDirectory(string directory)
+        {
+            if (string.IsNullOrEmpty(directory))
+            {
+                var currentDirectory = Directory.GetCurrentDirectory();
+                return NuGet.Configuration.Settings.LoadDefaultSettings(
+                   currentDirectory,
+                   configFileName: null,
+                   machineWideSettings: new XPlatMachineWideSetting());
+            }
+            if (Directory.Exists(directory))
+            {
+                return NuGet.Configuration.Settings.LoadDefaultSettings(
+                   directory,
+                   configFileName: null,
+                   machineWideSettings: new XPlatMachineWideSetting());
+            }
+
+            throw new CommandException(string.Format(CultureInfo.CurrentCulture, Strings.Error_PathNotFound, directory));
```

Comment:
```suggestion
            if (string.IsNullOrEmpty(directory))
            {
                directory = Directory.GetCurrentDirectory();
            }
            
            if (!Directory.Exists(directory))
            {
                throw new CommandException(string.Format(CultureInfo.CurrentCulture, Strings.Error_PathNotFound, directory));
            }

            return NuGet.Configuration.Settings.LoadDefaultSettings(
                       directory,
                       configFileName: null,
                       machineWideSettings: new XPlatMachineWideSetting());
```

Sorry now that this method was refactored, I see more improvements.